docs/streams/developer-guide/dsl-api.html

```diff
@@ -2969,6 +2971,62 @@ <h4><a id="streams_concepts_globalktable" href="#streams_concepts_globalktable">
 t=5 (blue), which lead to a merge of sessions and an extension of a session, respectively.</span></p>
                         </div>
                     </div>
+		    <div class="section" id="window-final-results">
+			    <span id="windowing-final-results"></span><h5><a class="toc-backref" href="#id31">Window Final Results</a><a class="headerlink" href="#window-final-results" title="Permalink to this headline"></a></h5>
+			    <p>In Kafka Streams, windowed computations update continuously.
+			       As new data arrives for a window, freshly computed results are emitted downstream.
+			       For most applications, this is ideal, since fresh results are always available with a minimum of configuration,
+			       and Kafka Streams is designed to make programming continuous computations seamless.
+			       However, some applications need to take action <strong>only</strong> on the final result of a windowed computation.
+			       A common example of this is sending alerts.
+			    </p>
+			    <p>Suppose that you have an hourly windowed count of events per user.
+			       If you wish to send an alert when a user has more than three events in an hour, you would need some state tracking
+			       to make sure that you don't send two alerts when a user has four events in one hour.
+			       This is a little cumbersome, but doable.
+			    </p>
+			    <p>On the other hand, if you want to send an alert when a user has <em>less than</em> three events in an hour, you have a real challange.
+			       This is fundamentally because all users would match this condition at first, until they accrue enough events, so you cannot simply
+			       send an alert when someone matches the condition; you have to wait until you know you won't see any more events for a particular window
+			       and <em>then</em> send the alert.
+			    </p>
+                            <p>Kafka Streams offers a cleaner way to support both of these use cases: after defining your windowed computation, you can
```

Comment:
The first case is not a perfect example, because one would want to send and alert when the count is reached and not wait for the window end?

Maybe simplify the example and only talk about "external side effect" that should not happen on each update? Also note, even with EOS, "external side effects" could be executed multiple times. Thus, it might even be better to be more abstract and use "downstream systems, that cannot handle updates" as an example?

Reply:
The first example is still applicable, since you have to set up a custom processor and state store to accomplish the task, but with suppression you can just wait for the "final results" and then take action. You wind up delaying the action in exchange for a simpler program... But if you think this is distracting, I can remove it.

Personally, I think a more concrete example is easier to digest. Especially since, during the design phase, all of us had trouble imagining what kind of external system can't handle updates. I thought that sending alerts, which someone mentioned specifically, really made the case.

I don't think that bringing up the fact that you can still get multiple executions with EOS simplifies the example at all, but it might still be an important point. On the other hand, this is already explained quite clearly in the docs/blogs about EOS. Given that there's nothing different about the situation in the context of suppression, is it distracting to mention it here? I could imagine writing a more narrative blog article that reminds people about this.

Reply:
I agree that a concrete and simple example would be best.

> all of us had trouble imagining what kind of external system can't handle updates.

Really? Some people might use Connect to put the result topic into HDFS -- no updates supported there.

Ack on EOS -- we can just omit it. This would lead down into the rabbit hole.